### PR TITLE
imagemagick: make 7.0 default

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/40015

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
@matthiasbeyer @aanderse There are patches in the cataract git repo to make it work with imagemagick7. Can you look into that?

File: pkgs/development/libraries/libdmtx/default.nix [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
 
   meta = {
     description = "An open source software for reading and writing Data Matrix barcodes";
-    homepage = "http://libdmtx.org";
+    homepage = "https://github.com/dmtx/libdmtx";
+    changelog = "https://github.com/dmtx/libdmtx/blob/v${version}/ChangeLog";
     platforms = lib.platforms.all;
     maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libdmtx/default.nix:26]: Maybe you can add yourself as a maintainer?

File: pkgs/development/libraries/vapoursynth/default.nix [hunk 2]
```diff
@@ -3,11 +3,11 @@
 , zimg, libass, python3, libiconv
 , ApplicationServices
 , ocrSupport ?  false, tesseract ? null
-, imwriSupport? true,  imagemagick7 ? null
+, imwriSupport? true,  imagemagick ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/vapoursynth/default.nix:6]:
```suggestion
, imwriSupport ? true,  imagemagick
```

File: pkgs/development/libraries/vapoursynth/default.nix [hunk 3]
```diff
@@ -3,11 +3,11 @@
 , zimg, libass, python3, libiconv
 , ApplicationServices
 , ocrSupport ?  false, tesseract ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/vapoursynth/default.nix:5]:
```suggestion
, ocrSupport ? false, tesseract
```

File: pkgs/development/libraries/vapoursynth/default.nix [hunk 4]
```diff
@@ -3,11 +3,11 @@
...
-, imwriSupport? true,  imagemagick7 ? null
+, imwriSupport? true,  imagemagick ? null
 }:
 
 assert ocrSupport   -> tesseract != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/vapoursynth/default.nix:9]:
```suggestion
```

File: pkgs/development/libraries/vapoursynth/default.nix [hunk 5]
```diff
@@ -3,11 +3,11 @@
...
-assert imwriSupport -> imagemagick7 != null;
+assert imwriSupport -> imagemagick != null;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/vapoursynth/default.nix:10]:
```suggestion
```

File: pkgs/development/libraries/libdmtx/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/libraries/libdmtx/default.nix:26]: No thanks.

**Comment by dotlambda** [comment]
This breaks `perlPackages.Gtk3ImageView`'s tests. cc @pacien

**Comment by matthiasbeyer** [comment]
> There are patches in the cataract git repo to make it work with imagemagick7. Can you look into that?

Yeah, well there was no release for cataract in 10 years, so I doubt there will be one anytime soon. If there is, we can update, but right now I don't think it is worth the effort if we can just use old imagemagick.

**Comment by pacien** [comment]
> This breaks `perlPackages.Gtk3ImageView`'s tests. cc @pacien

I've opened an issue upstream for the failing tests: https://github.com/carygravel/gtk3-imageview/issues/19.
Since Imagemagick is only used in the tests of Gtk3ImageView, I think it is safe to simply disable the tests of this package for now.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114007` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>imagemagick</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
imagemagick:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/graphics/ImageMagick/7.0.nix:83:3:

```
   |
83 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
</details>

**Pull request merged**